src/ga/ga_scaling.rs

```diff
@@ -1,11 +1,107 @@
-// TODO: COPYRIGHT, USE & AUTHORS
+// Copyright 2016 Revolution Solid & Contributors.
+// author(s): sysnett, carlos-lopez-garces
+// rust-monster is licensed under an MIT License.
+
+//! GA Scaling Schemes
+//!
+//! Scales the raw score of a Population's individuals.
+
 use super::ga_core::GASolution;
+use super::ga_population::GAPopulation;
 
 /// Scaling Scheme Trait
-///
-/// Interface to a Scaling Scheme. Goes from 'raw' objective values
-/// to fitness values.
+/// 
+/// Embeeded in the population, scales the values of raw score in a
+/// GASolution to set their fitness score
 pub trait GAScaling<T: GASolution>
 {
-    fn evaluate(pop : &mut GAPopluation<T>);
+    fn evaluate(&self, pop: &mut GAPopulation<T>);
+}
+
+/// No Scaling - Raw and Scaled are the same
+pub struct GANoScaling;
+
+impl<T: GASolution> GAScaling<T> for GANoScaling
+{
+    fn evaluate(&self, pop: &mut GAPopulation<T>)
+    {
+        // TODO: This is why we need iterators :(
+        let pop_vec = pop.population();
+        for ref mut ind in pop_vec
```

Comment:
`for ref mut ind in pop_vec` can also be `for ind in &mut pop_vec`. Do you know which form is idiomatic? Should we stick to one of them?

The 2 forms render the same results. Example:

```
pub struct S
{
    num: u32
}

fn main() {
    let mut v: Vec<S> = vec!(S{num: 1}, S{num: 2}, S{num: 3});
    for s in &mut v
    {
        s.num = s.num+10;
        println!("{}", s.num);
    }
    for ref mut s in v
    {
        s.num = s.num+10;
        println!("{}", s.num);
    }
}
```
Result:
11
12
13
21
22
23